pkg/agent/controller/egress/egress_controller_test.go

```diff
@@ -1604,3 +1917,51 @@ func TestCompareEgressStatus(t *testing.T) {
 		})
 	}
 }
+
+func TestEgressControllerReplaceEgressIPs(t *testing.T) {
+	c := newFakeController(t, []runtime.Object{
+		&crdv1b1.Egress{
+			ObjectMeta: metav1.ObjectMeta{Name: "egressA", UID: "uidA"},
+			Spec:       crdv1b1.EgressSpec{EgressIP: fakeLocalEgressIP1, ExternalIPPool: fakeExternalIPPool},
+			Status:     crdv1b1.EgressStatus{EgressNode: fakeNode, EgressIP: fakeLocalEgressIP1},
+		},
+		&crdv1b1.Egress{
+			ObjectMeta: metav1.ObjectMeta{Name: "egressB", UID: "uidB"},
+			Spec:       crdv1b1.EgressSpec{EgressIP: fakeLocalEgressIP2, ExternalIPPool: fakeExternalIPPool},
+			Status:     crdv1b1.EgressStatus{EgressNode: fakeNode, EgressIP: fakeLocalEgressIP2},
+		},
+		// Should not be included.
+		&crdv1b1.Egress{
+			ObjectMeta: metav1.ObjectMeta{Name: "egressC", UID: "uidA"},
+			Spec:       crdv1b1.EgressSpec{EgressIP: fakeRemoteEgressIP1, ExternalIPPool: fakeExternalIPPool},
+			Status:     crdv1b1.EgressStatus{EgressNode: fakeNode2, EgressIP: fakeRemoteEgressIP1},
+		},
+		&crdv1b1.Egress{
+			ObjectMeta: metav1.ObjectMeta{Name: "egressD", UID: "uidA"},
+			Spec:       crdv1b1.EgressSpec{EgressIP: "1.2.3.4", ExternalIPPool: "other-pool"},
+			Status:     crdv1b1.EgressStatus{EgressNode: fakeNode, EgressIP: "1.2.3.4"},
+		},
+		&crdv1b1.ExternalIPPool{
+			ObjectMeta: metav1.ObjectMeta{Name: fakeExternalIPPool, UID: "pool-uidA"},
+			Spec:       crdv1b1.ExternalIPPoolSpec{SubnetInfo: &crdv1b1.SubnetInfo{Gateway: fakeGatewayIP, PrefixLength: 16, VLAN: 2}},
+		},
+		&crdv1b1.ExternalIPPool{
+			ObjectMeta: metav1.ObjectMeta{Name: "other-pool", UID: "pool-uidB"},
+			Spec:       crdv1b1.ExternalIPPoolSpec{},
+		},
+	})
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	go c.podUpdateChannel.Run(stopCh)
+	c.crdInformerFactory.Start(stopCh)
+	c.informerFactory.Start(stopCh)
+	c.crdInformerFactory.WaitForCacheSync(stopCh)
+	c.informerFactory.WaitForCacheSync(stopCh)
+
+	c.mockIPAssigner.EXPECT().InitIPs(map[string]*crdv1b1.SubnetInfo{
+		fakeLocalEgressIP1: {Gateway: fakeGatewayIP, PrefixLength: 16, VLAN: 2},
```

Comment:
I didn't know that you could do this for map initialization (instead of `&crdv1b1.SubnetInfo{....}`). That's pretty cool.